console/v1/types_console_link.go

```diff
@@ -19,6 +19,16 @@ type ConsoleLinkSpec struct {
 	Link `json:",inline"`
 	// location determines which location in the console the link will be appended to.
 	Location ConsoleLinkLocation `json:"location"`
+	// applicationMenu holds information about section and icon used for the link in the applicationMenu.
+	ApplicationMenu ApplicationMenuSpec `json:"applicationMenu,omitempty"`
+}
+
+// ApplicationMenuSpec is the specification of the desired section and icon used for the link in the applicationMenu.
+type ApplicationMenuSpec struct {
+	// section is the section of the ApplicationMenu in which the link should appear.
+	Section string `json:"section"`
+	// imageUrl is the URL for the icon used in front of the link in the applicationMenu.
```

Comment:
I would explicitly say this can be an HTTPS URL or a Data URL. (It really should be HTTPS since console is always HTTPS, and browsers won't show mixed content.)

We should give some guidance on size and aspect ratio.

Reply:
> We should give some guidance on size and aspect ratio.

are there any estimation on any of these values ?
size - 32kb
ratio - 40x40

Reply:
The UI will show a 24x24 pixel image.